### DISCUSSION TITLE
Draw purge line behind the print

### DISCUSSION THREAD

**Discussion by Kall0** [jschuh/klipper-macros#184] [Q&A]
Hi! This is an awesome addition to Klipper, so thanks very many!

Question is as stated in the topic. Is there a way I could print the prime and purge section behind the actual print instead of at the front?

Issue is, that I would like to purge prime, but it always leaves a small blob on the print bed and Ender 3 s1 fan keeps grinding it and moves that blop usually in to the print. If it very behind the print area the fan wouldn't hit it. Any change here?

**Comment by kilinccagatay** [comment]
Hi, I was having the same issue with my sprite extruder upgraded E3V2 and I added this line to printer.cfg file changed the variable to "variable_start_purge_prime_length: 4" this worked for me and it is not creating a blob right now.

**Comment by Kall0** [comment]
Yea. I now changed to a totally different macro that purges on the side of the bed and then brushes against a brass brush I installed and then continues only to draw the purge lines and not prime at the point. Works pretty well. Still it would be nice to get the purge lines behind the print.

**Comment by jschuh** [comment]
Duplicate of #197

**Comment by jschuh** [comment]
On second thought, I'll respond here instead of #197.

I've never hit this because I guess I purge less. So, changing `variable_start_purge_prime_length` is an option, or setting `variable_start_purge_clearance` to move the whole purge line farther away from the print.

I could also look at doing the initial purge a bit differently, so it doesn't blob up so much (i.e. start it out a bit farther and move the X a bit during the initial purge).

**Comment by jschuh** [comment]
I changed the purge priming a bit in 2d77d095c69c77ca6d5cd919fe3ef870ef61a29a. Instead of just priming in place it now moves slowly in the X direction so it should blob significantly less. Although, the best way to address this is probably still to lower `variable_start_purge_prime_length`.
